clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,745 @@
+    with open(rules_json_file) as rules_file:
+        rules_object = json.load(rules_file)
+
+    if "inbound_rules" in rules_object.keys():
+        for rule in rules_object["inbound_rules"]:
+            ec2_add_inbound_security_rule(
+                ec2_client=ec2_client, security_grp_id=security_grp_id, sg_rule=rule
+            )
+
+    if "outbound_rules" in rules_object.keys():
+        for rule in rules_object["outbound_rules"]:
+            ec2_add_outbound_security_rule(
+                ec2_client=ec2_client, security_grp_id=security_grp_id, sg_rule=rule
+            )
+
+
+def create_public_subnets_for_vpc(
+    ec2_client, subnet_count, vpc_id, subnet_cidr_blocks, availability_zones, site_name
+) -> list:
+    """
+    creates public subnets in a given vpc
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subnet_count (int): The number of additional bits with which to extend the prefix
+         vpc_id (str): Id for the VPC where subnets need to be created
+         subnet_cidr_blocks (list): list of subnets CIDRs available
+         availability_zones (list): list of available zones
+         site_name (str): The value for 'Name' tag used for all resource
+
+    Returns:
+         list: list of ID's for the subnets created
+    """
+    LOGGER.info(f"Creating {subnet_count} Public Subnets...")
+    public_subnet_ids_list = []
+    for count in range(subnet_count):
+        subnet = ec2_client.create_subnet(
+            CidrBlock=subnet_cidr_blocks[count],
+            VpcId=vpc_id,
+            AvailabilityZoneId=availability_zones[count],
+        )
+        subnet_details = ec2_client.describe_subnets(
+            SubnetIds=[subnet["Subnet"]["SubnetId"]]
+        )
+        while True:
+            state = subnet_details["Subnets"][0]["State"]
+            if state == "available":
+                break
+            time.sleep(5)
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=subnet["Subnet"]["SubnetId"],
+            tag_name=f"{site_name}-subnet-public-{count+1}",
+        )
+        public_subnet_ids_list.append(subnet["Subnet"]["SubnetId"])
+    return public_subnet_ids_list
+
+
+def create_private_subnets_for_vpc(
+    ec2_client, subnet_count, vpc_id, subnet_cidr_blocks, availability_zones, site_name
+) -> list:
+    """
+    creates public subnets in a given vpc
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subnet_count (int): The number of additional bits with which to extend the prefix
+         vpc_id (str): Id for the VPC where subnets need to be created
+         subnet_cidr_blocks (list): list of subnets CIDRs available
+         availability_zones (list): list of available zones
+         site_name (str): The value for 'Name' tag used for all resource
+
+    Return:
+         list: list of ID's for the subnets created
+    """
+    LOGGER.info(f"Creating {subnet_count} Private Subnets...")
```

Comment:
```suggestion
    LOGGER.info(f"Creating {subnet_count} Private Subnets.")
```
please remove all extra dots in all places

Reply:
Done